src/libraries/System.Private.CoreLib/src/System/Globalization/CultureInfo.cs

```diff
@@ -496,6 +496,28 @@ public virtual CultureInfo Parent
                     CultureInfo culture;
                     string parentName = _cultureData.ParentName;
 
+                    if (parentName == "zh" && _name.Length == 5 && _name[2] == '-')
+                    {
+                        // We need to keep the parent chain for the zh cultures as follows to preserve the resource lookup compatability
+                        //      zh-CN -> zh-Hans -> zh -> Invariant
+                        //      zh-HK -> zh-Hant -> zh -> Invariant
+                        //      zh-MO -> zh-Hant -> zh -> Invariant
+                        //      zh-SG -> zh-Hans -> zh -> Invariant
+                        //      zh-TW -> zh-Hant -> zh -> Invariant
+
+                        if ((_name[3] == 'C' && _name[4] == 'N' ) || // zh-CN
```

Comment:
Was NLS case sensitive for these? I assume so

Reply:
We always normalize the culture names to such casing.